File: classic puzzle-easy/blowing-fuse.cpp
//https://www.codingame.com/training/easy/blowing-fuse
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    int m;
    int c;
    cin>>n>>m>>c;
    bool status[100]={false};
    int cc[100]={0};
    memset(status,false,n);
    for(int i=0;i<n;i++) cin>>cc[i];
    int current=0;
    int maxi_current=0;
    for (int i=0;i<m;i++){
        int mx;
        cin>>mx;
        mx-=1;
        if(!status[mx]){
            current+=cc[mx];
            if(current>c){
                cout<<"Fuse was blown."<<endl;
                exit(0);
            }
            status[mx]=!status[mx];
            if(current>maxi_current)
                maxi_current=current;
        }else{
            status[mx]=!status[mx];
            current-=cc[mx];
        }
    }

    cout << "Fuse was not blown." << endl;
    cout << "Maximal consumed current was "<<maxi_current<<" A." << endl;
}

File: classic puzzle-easy/blowing-fuse.py
#https://www.codingame.com/training/easy/blowing-fuse
#BLOWING FUSE
import sys
import math

n,m,c=[int(i) for i in input().split()]
status=[False]*n
cc=[int(i) for i in input().split()]
current=0
max_current=0
for i in input().split():
    mx=int(i)-1
    if not status[mx]:
        current+=cc[mx]
        if current>c:
            print("Fuse was blown.")
            exit()
        if current>max_current:
            max_current=current
        status[mx]=not status[mx]
    elif status[mx]:
        status[mx]=not status[mx]
        current-=cc[mx]
    
print("Fuse was not blown.")
print(f"Maximal consumed current was {max_current} A.")
